src/Classes/CommentDisplay.php

```diff
@@ -0,0 +1,351 @@
+<?php
+/**
+ * Curse Inc.
+ * Curse Profile
+ * A modular, multi-featured user profile system.
+ *
+ * @package   CurseProfile
+ * @author    Noah Manneschmidt
+ * @copyright (c) 2013 Curse Inc.
+ * @license   GPL-2.0-or-later
+ * @link      https://gitlab.com/hydrawiki
+ */
+
+namespace CurseProfile\Classes;
+
+use Html;
+use HydraCore;
+use MediaWiki\MediaWikiServices;
+use MediaWiki\Session\CsrfTokenSet;
+use Parser;
+use RequestContext;
+use SpecialPage;
+use Title;
+use User;
+
+/**
+ * A class to manage displaying a list of friends on a user profile
+ */
+class CommentDisplay {
+	/**
+	 * Responds to the comments parser hook that displays recent comments on a profile
+	 *
+	 * @param mixed &$parser parser instance
+	 * @param int $userId id of the user whose recent comments should be displayed
+	 * @return array with html at index 0
+	 */
+	public static function comments( &$parser, int $userId ) {
+		global $wgUser;
+
+		if ( $userId < 1 ) {
+			return 'Invalid user ID given';
+		}
+
+		$userFactory = MediaWikiServices::getInstance()->getUserFactory();
+		$html = '';
+
+		$boardOwner = $userFactory->newFromId( $userId );
+		$html .= self::newCommentForm( $boardOwner, false );
+
+		$board = new CommentBoard( $userFactory->newFromId( $userId ) );
+		$comments = $board->getComments( $wgUser );
+
+		foreach ( $comments as $comment ) {
+			$html .= self::singleComment( $comment, false );
+		}
+
+		return [
+			$html,
+			'isHTML' => true,
+		];
+	}
+
+	/**
+	 * Returns the HTML text for a comment entry form if the current user is logged in and not blocked
+	 *
+	 * @param User $owner User whose comment board will receive a new comment via this form
+	 * @param bool $hidden If true, the form will have an added class to be hidden by css/
+	 * @return string html fragment or empty string
+	 */
+	public static function newCommentForm( User $owner, bool $hidden = false ) {
+		global $wgRequest, $wgUser;
```

Comment:
$wgUser is deprecated, though it's a soft-deprecation - so we can skip it for now if it'd be too much work. Ideally we'd pass down an `IContextSource` down the call chain from somewhere where we can obtain a non-main one (e.g. from a special page).